### ISSUE TITLE
Learning rate related issues

### ISSUE THREAD

**Issue by 1978679623** [lyq133/LIC-HPCM#14]
Hello, thank you for sharing your project code. I would like to ask about the adjustment of the learning rate in this; the one in the paper is a bit different from the one in the code.
In the paper, it is 
The learning rate is set as 10−4 initially, and reduced to 2 × 10−5 after 1.6M steps, then to 5 × 10−6 after 1.8M steps, and then to 10−6 after 1.9M steps.
Inside the code is
lr_scheduler = lambda x : \
1e-4 if x < 2750 else (
    3e-5 if x < 2850 else (
        1e-5 if x < 2950 else 1e-6
    )
)
The learning rate in this code is different from the two intermediate values of the learning rate in the paper. Also, may I ask how the numbers after x are obtained? Is it based on the number of training steps? After 1.6M steps, the learning rate becomes 2 × 10−5; after 1.8M steps, it becomes 5 × 10−6; and after 1.9M steps, it becomes 10−6. However, my own calculations are different from the numbers you set. Could you help me understand how 2750, 2850, 2950 were obtained? If the number of training epochs is different, do these numbers need to be adjusted accordingly? I am really looking forward to your answer. Thank you.

**Comment by lyq133** [comment]
Hi, thanks for your interest in our work.

In the released code, the values 2750, 2850, and 2950 refer to epoch milestones rather than the exact number of training iterations. They are approximate settings used in our training script. The learning-rate schedule in the code is therefore a slightly simplified/approximate version of the schedule described in the paper, and the difference in the two learning rates should have little influence on the final performance.

If you would like to strictly follow the paper, you can adjust the milestones according to the actual number of iterations per epoch in your training setting, or implement the learning-rate decay based on the global training step directly. If the total number of epochs, dataset size, or batch size is changed, these epoch milestones should also be adjusted accordingly.

Thanks again for pointing this out.

**Comment by 1978679623** [comment]
Thank you for your reply. I wish you smooth work and enduring academic success.
